src/views/AddItem.jsx

```diff
@@ -1,3 +1,111 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useEffect, useState } from 'react';
+import { addItem } from '../api/firebase';
+
+export function AddItem({ listId }) {
+	const [timeframe, setTimeframe] = useState('7');
+	const [itemName, setItemName] = useState('');
+	const [message, setMessage] = useState(null);
+
+	useEffect(() => {
+		const timer = setTimeout(() => {
+			setMessage(null);
+		}, 3000);
+
+		return () => clearTimeout(timer);
+	}, [message]);
+
+	const onTimeChange = (e) => setTimeframe(e.target.value);
+	const onItemChange = (e) => setItemName(e.target.value);
+	const onFormSubmit = async (e) => {
+		e.preventDefault();
+		// Make sure the user has entered an item name
+		if (!itemName) {
+			setMessage('Please enter an item name');
+			return;
+		}
+
+		const result = await addItem(listId, {
+			itemName,
+			daysUntilNextPurchase: timeframe,
+		});
+		if (result) {
+			setMessage(`Added ${itemName} to your list.`);
+			setItemName('');
+			setTimeframe(7);
```

Comment:
I don't think you're getting the desired effect of the button being reset to "soon" because of the differences in types. The timeframe is initialized as a string, but here it's being reset to a number. This is why you're getting the behaviour that @lyleschemmerling showed in the video

Reply:
This was exactly it, should be fixed now @lyleschemmerling. Thank you @mentalcaries.

Reply:
Good job! Form resets after submit